02-intermediate-ruby/api-auth.md

```diff
@@ -1,3 +1,226 @@
 # APIs and Authorization
 
-# Pull from intro to github OAuth (.env, .gitignore)
\ No newline at end of file
+## Learning Goals
+
+By the end of this lesson, students should be able to...
+
+- Use an API that requires an _authentication token_
+- Store _API credentials_ securely
+- Recognize and use _environment variables_ in Ruby with the `dotenv` gem
+
+## Authentication
+
+While some APIs are open for anyone to use, many require you to have an account and to identify yourself when you send a request.
+
+Requiring users to have an account has many benefits for API maintainers:
+
+- Improved data about users
+- Ability to rate-limit or "throttle" requests
+- Ability to charge users for certain features
+
+_Authentication_ is a branch of computer security that deals with the questions "who are you and how can I trust you".
+
+Authentication is distinct from _authorization_, which answers the question "what am I allowed to do?". Both start with "auth", which can be a little confusing. Here's a way to remember which is which:
```

Comment:
Small suggestions: Instead of comparing authentication/authorization AND defining authorization in the same sentence, maybe split this into two sentences:

Authentication is distinct from _authorization_. Authorization answers the question "what am I allowed to do?".